autoload/ale/statusline.vim

```diff
@@ -14,34 +14,48 @@ function! ale#statusline#Update(buffer, loclist) abort
         endif
     endfor
 
-    let g:ale_buffer_count_map[a:buffer] = [l:errors, l:warnings]
+    let g:ale_buffer_info[a:buffer].count = [l:errors, l:warnings]
 endfunction
 
-" Returns a tuple of errors and warnings for use in third-party integrations.
-function! ale#statusline#Count(buffer) abort
+" Set the error and warning counts, calling for an update only if needed.
+" If counts cannot be set, return 0.
+function! s:SetupCount(buffer) abort
+    if !has_key(g:ale_buffer_info, a:buffer)
+        " Linters have not been run for the buffer yet, so stop here.
+        return 0
+    endif
+
     " Cache is cold, so manually ask for an update.
-    if !has_key(g:ale_buffer_count_map, a:buffer)
+    if !has_key(g:ale_buffer_info[a:buffer], 'count')
         call ale#statusline#Update(a:buffer, get(g:ale_buffer_loclist_map, a:buffer, []))
```

Comment:
I'm considering removing that global entirely. I'm in two minds about it. It's kind of an easy format to read in case anyone wants to use it for something or other. Then again, maybe it's only useful in this plugin itself.